src/users.ts

```diff
@@ -0,0 +1,901 @@
+import Errors from './errors'
+
+export interface User {
+  name?: string
+  id?: string
+  credentials?: {
+    identity: string
+    secret?: string
+  }
+  owner?: string
+  tags?: [string, string]
+  role?: string
+  status?: 'enabled' | 'disabled'
+  createdAt?: string
+  updatedAt?: string
+  updatedBy?: string
+}
+
+interface PageRes {
+  total: number
+  offset: number
+  limit: number
+}
+
+interface UsersInterface {
+  users: User[]
+  page: PageRes
+}
+
+interface Group {
```

Comment:
These interfaces should be imported from their respective files not created here and then being created in the groups file again. Same case for things, channels, and groups.

Reply:
Or we can have a common `defs.ts` file where all the interfaces are defined and  other files can import these interfaces from therer

Reply:
since this is just a users.ts PR defining the interfaces here should be okay since the other files are still in .js format. Let me add a defs.ts file instead for more structure.